freqtrade/resolvers/hyperopt_resolver.py

```diff
@@ -45,12 +45,12 @@ def load_hyperopt(config: Dict) -> IHyperOpt:
         if not hasattr(hyperopt, 'populate_indicators'):
             logger.info("Hyperopt class does not provide populate_indicators() method. "
                         "Using populate_indicators from the strategy.")
-        if not hasattr(hyperopt, 'populate_buy_trend'):
-            logger.info("Hyperopt class does not provide populate_buy_trend() method. "
-                        "Using populate_buy_trend from the strategy.")
-        if not hasattr(hyperopt, 'populate_sell_trend'):
-            logger.info("Hyperopt class does not provide populate_sell_trend() method. "
-                        "Using populate_sell_trend from the strategy.")
+        if not hasattr(hyperopt, 'populate_enter_trend'):
+            logger.info("Hyperopt class does not provide populate_enter_trend() method. "
+                        "Using populate_enter_trend from the strategy.")
+        if not hasattr(hyperopt, 'populate_exit_trend'):
+            logger.info("Hyperopt class does not provide populate_exit_trend() method. "
+                        "Using populate_exit_trend from the strategy.")
```

Comment:
Compatibility code required

Reply:
What do you mean by compatibility code?

Reply:
well - code that ensures a strategy  (or in this case, hyperopt file) that implements `populate_sell_trend` - but not `populate_exit_trend` will continue to work as it used to.

Reply:
> well - code that ensures a strategy (or in this case, hyperopt file) that implements `populate_sell_trend` - but not the new `populate_exit_trend` will continue to work as it used to (making it compatible with the old version, too).

What do you think the best way to do that would be?

Reply:
i'd start with the strategy - maybe look at the code in `advise_buy`/`advise_sell` - which also has some level of compatibility code (we've added a new parameter in the past).

In theory, you can detect if an object has the old but not the new method, and eventually simply reassing (a function is just an object in python) - or call the old function instead if you detect that case (make sure to detect the change ONCE, and not on every call).

Hyperopt might be more complex - as there it seems we're assigning `advise_buy` / `advise_sell` directly (which would need further investigation if that's really necessary or a mistake).